aws-py-eks/__main__.py

```diff
@@ -2,38 +2,43 @@
 import vpc
 import utils
 import pulumi
+import pulumi_aws as aws
 from pulumi_aws import eks
 
-## EKS Cluster
+provider = aws.Provider('provider', region='us-west-2')
 
-eks_cluster = eks.Cluster(
-    'eks-cluster',
-    role_arn=iam.eks_role.arn,
-    tags={
-        'Name': 'pulumi-eks-cluster',
-    },
-    vpc_config=eks.ClusterVpcConfigArgs(
-        public_access_cidrs=['0.0.0.0/0'],
-        security_group_ids=[vpc.eks_security_group.id],
-        subnet_ids=vpc.subnet_ids,
-    ),
-)
+## EKS Cluster
+with pulumi.default_providers([provider]):
+    ec2_role, eks_role = iam.setup_iam()
+    eks_security_group, subnet_ids = vpc.vpc()
```

Comment:
This is sort of the interesting bit - and also where "global" approach would run into some problems - I assume the dependent modules would all be loaded and run before we could "install" the default provider?

That said - this does lead to a decent amount more code refactoring to get to something that works?